integration-cli/docker_cli_daemon_test.go

```diff
@@ -481,6 +481,31 @@ func TestDaemonUpgradeWithVolumes(t *testing.T) {
 	logDone("daemon - volumes from old(pre 1.3) daemon work")
 }
 
+// GH#11320 - verify that the daemon exits on failure properly
+func TestDaemonExitOnFailure(t *testing.T) {
```

Comment:
Why add such a specific comment for this? We're testing that the engine is shutdown and that docker exits when there is a misconfiguration of any kind causing the bootup to fail.